### ISSUE TITLE
Feature idea: Directory contents

### ISSUE THREAD

**Issue by adampresley** [mjibson/esc#47]
Let me preface my feature request by first stating that I really have enjoyed using **esc**. It is simple, uses the Go toolchain (generate), and makes me happy.

I have run across a use case where a possible feature enhancement could make this tool a little easier to use for specific use cases. Below is a sample of some code that demonstrates a scenario where I have implemented a renderer for use in the Echo framework. Simply put it provides a **Render** method that tells the framework how to render requests. 

In the **LoadTemplates** method is where the imaginary code comes in. In here I imagine the ability to do something similar to *ioutil.ReadDir()* and get a slice of **FileInfo** structs. This way I can then iterate over the files and compile them as templates.

```
package ui

import (
	"fmt"
	"html/template"
	"io"
	"path/filepath"
	"strings"

	"myproject/app/compiledassets"
	"github.com/labstack/echo"
)

var templates map[string]*template.Template

/*
TemplateRenderer describes a handlers for rendering layouts/pages
*/
type TemplateRenderer struct {
	templates *template.Template
}

/*
NewTemplateRenderer creates a new struct
*/
func NewTemplateRenderer(debugMode bool) *TemplateRenderer {
	result := &TemplateRenderer{}
	result.LoadTemplates(debugMode)

	return result
}

/*
Implements render interface expected by Echo 
*/
func (t *TemplateRenderer) Render(w io.Writer, name string, data interface{}, ctx echo.Context) error {
	var tmpl *template.Template
	var ok bool

	if tmpl, ok = templates[name]; !ok {
		return fmt.Errorf("Cannot find template %s", name)
	}

	return tmpl.ExecuteTemplate(w, "layout", data)
}

/*
Traverses the asset pages directory and compiles templates into a package level variable
*/
func (t *TemplateRenderer) LoadTemplates(debugMode bool) {
	templates = make(map[string]*template.Template)

	/*
	 * This code isn't real. It is what I imagine esc could look like
	 * to get a directory of assets. Perhaps returns a slice of 
	 * os.FileInfo structs
	 */
	files := compiledassets.DirContents("/www")

	for _, file := range files {
		if !file.IsDir() {
			basename := file.Name()
			trimmedName := strings.TrimSuffix(basename, filepath.Ext(basename))

			templates["mainLayout:"+trimmedName], _ = template.Must(
				template.New("layout").Parse(www.FSMustString(debugMode, "/www/myproject/layouts/mainLayout.gohtml")),
			).Parse(www.FSMustString(debugMode, "/www/myproject/pages/"+basename))
		}
	}
}
```

What are your thoughts on this? Thanks for reading!

**Comment by mjibson** [comment]
Esc already has a Readdir method, it just always returns nil. Since esc also keeps track of things that are directories, it should be possible to just correctly implement Readdir. I'm not going to do this work, but I would review a PR that did.

**Comment by adampresley** [comment]
OK. I may take a look at this and see what an implementation might look like. Thanks for responding!

**Comment by andig** [comment]
Looks like this has been implemented 1ee720efa6512a6b998fb92ce7306b506311d1aa. Would be great if the docs could be improved.

**Comment by adampresley** [comment]
It does look like it is implemented. I'll take a look at it, and may see about contributing a doc update for this.
